plugins/inputs/smart/smart_test.go

```diff
@@ -389,12 +395,18 @@ func TestGatherSATAInfo(t *testing.T) {
 	)
 
 	wg.Add(1)
-	gatherDisk(acc, internal.Duration{Duration: time.Second * 30}, true, true, "", "", "", wg)
+	s.gatherDisk(acc, "", wg)
```

Comment:
You probably should wait for the accumulator to collect the metrics using `acc.Wait()` otherwise this test might be racy.

Reply:
You are right, it's not my test I was just changing where the gatherDisk() comes from.